x-pack/solutions/observability/plugins/synthetics/server/tasks/sync_private_locations_monitors_task.ts

```diff
@@ -100,12 +105,34 @@ export class SyncPrivateLocationMonitorsTask {
         MAINTENANCE_WINDOW_SAVED_OBJECT_TYPE,
       ]);
 
+      const allPrivateLocations = await getPrivateLocations(soClient, ALL_SPACES_ID);
+      if (allPrivateLocations.length === 0) {
+        this.debugLog(`No private locations found, skipping sync of private location monitors`);
+        return {
+          state: taskState,
+          schedule: {
+            interval: TASK_SCHEDULE,
+          },
+        };
+      }
       const { performCleanupSync } = await this.cleanUpDuplicatedPackagePolicies(
         soClient,
         taskState
       );
 
-      const allPrivateLocations = await getPrivateLocations(soClient, ALL_SPACES_ID);
+      const hasMonitorsWithMWs = await this.monitorsHaveMaintenanceWindows(soClient);
+      if (!hasMonitorsWithMWs) {
+        this.debugLog(
+          `No monitors with maintenance windows found, skipping sync of private location monitors`
+        );
+        return {
+          state: taskState,
+          schedule: {
+            interval: TASK_SCHEDULE,
+          },
+        };
+      }
+
```

Comment:
We are changing the behaviour of the task, previously, we would have run `cleanUpDuplicatedPackagePolicies` and `hasMWsChanged` even if the user has no private location, but not anymore. 
Is that expected?